mos-platform/common/c/string.c

```diff
@@ -63,3 +63,70 @@ __attribute__((weak)) char *strrchr(const char *s, int c) {
       last = s;
   return (char *)last;
 }
+
+__attribute__((weak)) char* strrev(char* str)
+{
+    size_t len = strlen((const char*)str);
+    size_t i, j;
+    for (i = 0, j = len - 1; i < j; i++, j--)
+    {
+        char a = str[i];
+        str[i] = str[j];
+        str[j] = a;
+    }
+	return str;
+}
+
+__attribute__((weak)) size_t strspn(const char* string, const char* in)
+{
+    const char *s1, *s2;
+
+    for (s1 = string; *s1; s1++)
+    {
+        for (s2 = in; *s2 && *s2 != *s1; s2++)
+            ;
+        if (!*s2)
+            break;
+    }
+    return s1 - string;
+}
+
+__attribute__((weak)) char* strpbrk(const char* string, const char* brk)
+{
+    while (*string)
+    {
+        const char* s1;
+        for (s1 = brk; *s1 && *s1 != *string; s1++)
+            ;
+        if (*s1)
+            return (char*)string;
+        string++;
+    }
+    return (char*)NULL;
+}
+
+__attribute__((weak)) char* strtok(char* string, const char* separators)
+{
+    static char* savestring = NULL;
+
+    if (string == NULL)
+    {
+        string = savestring;
+        if (string == NULL)
+            return (char*)NULL;
+    }
+
+    char* s1 = string + strspn(string, separators);
+    if (*s1 == '\0')
```

Comment:
Use !*s1, for consistency with general use of *s1 as a conditional

Reply:
Done.